[Glenda Chong] iP

GlendaChong opened the pull request:
# GlendaPro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

GlendaPro frees your mind of having to remember things you need to do! It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://github.com/GlendaChong/ip)
2. double-click it
3. add your tasks 
4. let it manage your tasks for you 😄

And it is **FREE**!

Features: 
- [x] Managing todo tasks
- [x] Managing deadline tasks
- [x] Managing event tasks
- [x] List out all the tasks in the task list
- [x] Mark and unmark tasks as completed 
- [x] Delete tasks from task list
- [x] Find tasks based on keyword 
- [ ] Reminders (coming soon)

--- 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 
```java
public class Main {
    public static void main(String[] args) {
         Application.launch(MainApp.class, args); 
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

GlendaChong committed:
Rename chatbot, greet user, exit

GlendaChong committed:
Echo commands

GlendaChong committed:
Add and display tasks

GlendaChong committed:
Added mark task as done or undone feature

GlendaChong committed:
ToDos, Events, Deadlines

GlendaChong committed:
Automated Text UI Testing

GlendaChong committed:
Added error handling

GlendaChong committed:
Edited tasks to use ArrayList<>

GlendaChong committed:
Added delete task function

GlendaChong committed:
Add save tasks to hard disk functionality

GlendaChong committed:
Merge branch 'branch-Level-7'

GlendaChong committed:
Add date and time functionality for deadline tasks

GlendaChong committed:
Update automated text-ui testing

GlendaChong committed:
Merge branch 'branch-Level-8'

GlendaChong committed:
Add more OOP

GlendaChong committed:
Organise classes into packages

GlendaChong committed:
Merge branch 'branch-Packages'

GlendaChong committed:
Merge branch 'add-gradle-support'

GlendaChong committed:
Add JUnit tests

GlendaChong committed:
Merge branch 'branch-JUnit'

GlendaChong committed:
Edit build.gradle file

GlendaChong committed:
Create a JAR file

GlendaChong committed:
Add Javadocs to all current classes and methods

GlendaChong committed:
Follow coding standard and refactor code

GlendaChong committed:
Add find tasks by keyword search functionality

GlendaChong committed:
Merge branch 'branch-A-JavaDoc'

GlendaChong committed:
Merge branch 'branch-A-CodingStandard'

GlendaChong committed:
Merge branch 'branch-Level-9'

GlendaChong committed:
Fix event task displays in saved file and chatbot

GlendaChong committed:
Update styling of codes

GlendaChong committed:
Merge branch 'A-Checkstyle'

GlendaChong committed:
Add GUI to chatbot

GlendaChong committed:
Merge branch 'branch-Level-10'

spatuly commented on src/main/java/duke/command/InvalidCommand.java:
```diff
@@ -0,0 +1,22 @@
+    public InvalidCommand(DukeException error) {
+        this.error = error;
+    }
+    @Override
```
Consider adding a blank line between lines 17 and 18.

GlendaChong replied:
Will do, thank you!

spatuly commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * Displays the task added and the current total number of tasks in the list.
+     * @param task The task that was added.
```
I think there should be an empty line between description and params?

GlendaChong replied:
Thanks for pointing that out!

spatuly commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+ * A GUI for Duke using FXML.
+ */
+public class Main extends Application {
+    public static final String FILE_PATH = "data/duke.txt";
```
Should this be private instead?

GlendaChong replied:
Yep, thanks for pointing that out!

spatuly commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+    }
+
+    /**
+     * Loads the tasks from the file
```
Could consider adding a full stop?

spatuly commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    public TaskList() {
```
Maybe consider adding empty lines between line 10 and 11 as well as 13 and 14.

spatuly commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,59 @@
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+/**
```
Maybe add a blank line before line 11 for better consistency / readability.

GlendaChong replied:
Thanks for pointing that out!

spatuly commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,64 @@
+/**
+ * Duke is a personal assistant chatbot that helps
+ * to keep track of various tasks.
+ *
```
I think this line can be deleted?

spatuly reviewed (commented):
Looks good, easy to read and mostly adheres to the coding standard. Only very few very minor changes requested. Also might want to consider adding more blank lines to separate logical units in a block. Good job :)

Clin-lyx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,64 @@
+ *
+ */
+public class Duke {
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
```
Please CAP the final variables :)

GlendaChong replied:
Thanks for pointing that out!

Clin-lyx commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,30 @@
+ * The AddCommand class represents a command to add a new task to the task list.
+ */
+public class AddCommand extends Command {
+    private final Task task;
```
Please CAP task here :)

Clin-lyx commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,33 @@
+ * The DeleteCommand class represents a command to deleted from the task list.
+ */
+public class DeleteCommand extends Command {
+    private final int taskNumber;
```
I've noticed several instances of the same issue; please change the variable names of constants to uppercase letters.

Clin-lyx commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            return new AddCommand(new ToDo(splitCommand[1]));
+        case "mark":
+        case "unmark":
```
Maybe u can delete these two lines?

GlendaChong replied:
Hello, I don't think I should delete these 2 lines. They are not empty cases, but rather have the same method implementation as the "delete" case, hence they are placed together like this. Hope that clarifies!

Clin-lyx reviewed (commented):
Overall, I find the quality of your code to be quite high. Just some minor issues with the coding standard (mainly final variables) and some lengthy method that affects the readability.

GlendaChong committed:
Refactor code based on PR reviews

GlendaChong committed:
Merge branch 'branch-RefactorCode'

GlendaChong committed:
Add assertions to some files

Code does not have any assert functions. Assertions can indicate
mistakes made in the code.

Let's add assertions to Parser, Storage, Duke, Deadline, Event
and Task files.

GlendaChong committed:
Improve code quality

Let's
* remove overloaded Deadline constructor to simplify code
* make the happy path prominent
* abstract code out into functions to remove arrowhead

GlendaChong committed:
Merge pull request #1 from GlendaChong/branch-A-Assertions

Add assertions to some files

GlendaChong committed:
Merge pull request #2 from GlendaChong/branch-A-CodeQuality

Improve code quality

GlendaChong committed:
Set up Continuous Integration(CI)

GlendaChong committed:
Fix checkstyle and test errors

GlendaChong committed:
Update gradle.yml file to fix build error

GlendaChong committed:
Add unscheduled tasks with a fixed duration feature

GlendaChong committed:
Merge branch 'branch-B-FixedDurationTasks'

GlendaChong committed:
Improve the GUI to look more like a chatbot

Update profile pictures and add a textbox border around the text.

GlendaChong committed:
Merge branch 'branch-A-BetterGui'

GlendaChong committed:
Improve code quality and add screenshot of product

GlendaChong committed:
Add user guide

GlendaChong committed:
Merge branch 'branch-A-UserGuide'

GlendaChong committed:
Update user guide

GlendaChong committed:
Merge branch 'branch-A-UserGuide'

GlendaChong committed:
Update Getting Started instructions in README

GlendaChong committed:
Update README.md
